### PR TITLE
all/any for observables

### PR DESCRIPTION
**Pull request by azlobinjb**
We had an ad hoc 'any' implementation in SharingSettingsMapper. 'all' could also be useful.

### CODE REVIEW

File: model/src/main/java/jetbrains/jetpad/model/collections/list/ObservableCollections.java [hunk 1]
```diff
@@ -128,6 +129,69 @@ public void onEvent(CollectionItemEvent<? extends ItemT> event) {
     return EMPTY_LIST;
   }
 
+  private static <ItemT> ReadableProperty<Boolean> total(
```
Comment (glebleonov) [inline model/src/main/java/jetbrains/jetpad/model/collections/list/ObservableCollections.java]: Such implementation leads to O(collection size) update for single collection change. 
What do you think of returning Property<Integer> with number of items that satisfy given predicate?

**Comment by azlobinjb** [comment]
@glebleonov Please take a look

File: model/src/main/java/jetbrains/jetpad/model/collections/list/ObservableCollections.java [hunk 2]
```diff
@@ -128,6 +143,84 @@ public void onEvent(CollectionItemEvent<? extends ItemT> event) {
...
+          return simpleCount
+              (predicate, collection);
+        } else {
+          return myCount;
+        }
+      }
+    };
+  }
+
+  private static <ItemT> int simpleCount
```
Comment (glebleonov) [inline model/src/main/java/jetbrains/jetpad/model/collections/list/ObservableCollections.java]: unnecessary line break

File: model/src/main/java/jetbrains/jetpad/model/collections/list/ObservableCollections.java [hunk 3]
```diff
@@ -128,6 +143,84 @@ public void onEvent(CollectionItemEvent<? extends ItemT> event) {
...
+      @Override
+      protected void doRemoveListeners() {
+        myCollectionRegistration.remove();
+        myCollectionRegistration = null;
+      }
+
+      @Override
+      protected Integer doGet() {
+        if (myCollectionRegistration == null) {
+          return simpleCount
```
Comment (glebleonov) [inline model/src/main/java/jetbrains/jetpad/model/collections/list/ObservableCollections.java]: unnecessary line break

File: model/src/main/java/jetbrains/jetpad/model/collections/list/ObservableCollections.java [hunk 4]
```diff
@@ -128,6 +143,84 @@ public void onEvent(CollectionItemEvent<? extends ItemT> event) {
...
+
+          @Override
+          public void onItemRemoved(CollectionItemEvent<? extends ItemT> event) {
+            if (predicate.test(event.getOldItem())) {
+              myCount--;
+            }
+            somethingChanged();
+          }
+        });
+        myCount = simpleCount
```
Comment (glebleonov) [inline model/src/main/java/jetbrains/jetpad/model/collections/list/ObservableCollections.java]: unnecessary line break

**Comment by azlobinjb** [comment]
@glebleonov fixed formatting

**Pull request merged**